Code/Framework/AzFramework/AzFramework/IO/LocalFileIO.cpp

```diff
@@ -480,6 +480,11 @@ namespace AZ
                 return false;
             }
 
+            if (AZStd::string(path).starts_with("/APK/"))
+            {
+                path += 5;
+            }
```

Comment:
This is the hack. We have to find a way that Android doesn't sent requests to VFS that start with "/APK/".

Reply:
Android sets the asset path based on what [FindAssetsDirectory()](https://github.com/aws-lumberyard-dev/o3de/blob/56e7e7073518847665638777b735ca7a64cdf9cd/Code/Framework/AzCore/Platform/Android/AzCore/Android/Utils.cpp#L114) returns. That function looks for engine.json in the app public storage and if it doesn't find it there, it looks in the APK. Maybe your engine.json file got deployed to the APK?

Reply:
It feels like what should be happening is that the results of FindAssetsDirectory should be populating the alias... not being prepended to file paths.

Reply:
I think that's what ends up happening. The Android [`GetDefaultAppRootPath()`](https://github.com/aws-lumberyard-dev/o3de/blob/f648cb1fd88ceeacada17e827d944bf124a195d0/Code/Framework/AzCore/Platform/Android/AzCore/Utils/Utils_Android.cpp#L51) calls `FindAssetsDirectory()`. Launcher calls `GetDefaultAppRootPath()` [here](https://github.com/aws-lumberyard-dev/o3de/blob/09dd3a87299c20b7c27da52f8f866f7ea9b7098f/Code/LauncherUnified/Launcher.cpp#L358) and then sets the `project_path` and `engine_path` in the settings registry(it adds them as command-line params) immediately after that. This is later used to set the runtime file paths [here](https://github.com/aws-lumberyard-dev/o3de/blob/5e4e61c0da6e3bdc949a5733a4d94bd449696adf/Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp#L841). The cache path registry key is set a few lines below [here](https://github.com/aws-lumberyard-dev/o3de/blob/5e4e61c0da6e3bdc949a5733a4d94bd449696adf/Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp#L887). And then finally, the alias for `@products@` gets set [here](https://github.com/aws-lumberyard-dev/o3de/blob/34a7db4582f28153532c1bc759b6ac86f47a7116/Code/Framework/AzFramework/AzFramework/Application/Application.cpp#L714). I think in @moraaar's case, the product alias is getting set to "/APK" because the engine.json file is in the APK and not in the sdcard files folder.

Reply:
Thanks @amzn-sj I'll investigate how to copy the minimum files into the sdcard instead of the apk. Thanks for pointing out a few places in the code, I can debug them as well to see what's going on.